debezium-connector-mongodb/src/test/java/io/debezium/connector/mongodb/junit/MongoDbDatabaseProvider.java

```diff
@@ -63,7 +75,20 @@ public static MongoDbShardedCluster mongoDbShardedCluster() {
                 .shardCount(shardSize)
                 .replicaCount(replicaSize)
                 .routerCount(1)
-                .portResolver(portResolver)
+                .portResolver(portResolver);
+    }
+
+    public static MongoDbShardedCluster mongoDbShardedCluster() {
+        var cluster = mongoDbShardedClusterBuilder();
+        if (NETWORK != null) {
+            cluster.network(NETWORK);
+        }
```

Comment:
Why do you need this when there is also the method which takes a network?

Reply:
Because the common/main entrypoint is `io.debezium.connector.mongodb.junit.MongoDbDatabaseProvider#externalOrDockerReplicaSet(org.testcontainers.containers.Network)` which is there to support external `DocumentDB`.
I think this needs to be cleaned up?

Reply:
Unless you intend to test with Document DB (and in that case I'd like to see the actual test result, please). Then I don't see how that would be relevant. 

In case of external database the network is logically not needed at all. We certainly can revisit this to make thing more compact again. However I don't see a reason for these changes as part of this PR when for your intended purpose you can just call the specific method and provide the network directly.

Reply:
I assume it makes a lot of sense to test the sink against different providers eventually.